src/components/MapLayerAnomalies/MapLayerAnomalies.js

```diff
@@ -0,0 +1,148 @@
+import { LAYER_FADE_DURATION_MS } from '@/lib/constants'
+
+export default {
+  name: 'v-mapbox-anomalies-layer',
+
+  inject: ['getMap'],
+
+  render: () => null,
+
+  props: {
+    options: {
+      type: Object,
+      default: () => ({}),
+    },
+  },
+
+  computed: {
+    layerDate () {
+      return this.$store.getters['anomalies-layers/anomaliesDate']
+    },
+  },
+
+  methods: {
+    deferredMountedTo () {
+      if (!this.isInitialized) {
+        this.initalize()
+        this.isInitialized = true
+      }
+    },
+
+    initalize () {
+      if (this.layerId) {
+        this.addLayerByDate(this.layerDate)
+      }
+    },
+
+    layerId (layerDate) {
+      const [year, month] = layerDate.split('-')
+      return `concat_anomaly_${year}_${parseInt(month)}`
+    },
+
+    addLayerByDate (layerDate) {
+      const map = this.getMap()
+      if (!map) { return }
+      const { firstLayerDate, lastLayerDate, styles, clickFn } = this.options
+
+      if (layerDate < firstLayerDate || layerDate > lastLayerDate) {
+        console.warn('Layer date is out of range:', layerDate)
```

Comment:
What will the user see?

Reply:
The user doesn't have a way of selecting a layer that would fall here, so I will remove the warning